File: code/Java/Chapter3/3.6 反转链表/Inverse.java
import java.util.ArrayList;

public class Inverse {
    public static void main(String[] args) {
        LinkedList list = new LinkedList();

        int n = 6;
        for(int i = 0; i < n; i++) {
            list.insert(i, new Node(i));
        }

        inverse(list.getHead());
        System.out.println();
    }

    public static void inverse(Node head) {
        if(head != null) {
            inverse(head.next);
            System.out.print(head.data + " ");
        }
    }
}

File: code/C/Chapter3/3.6 反转链表/inverse.c
#include <stdio.h>
#include "linked_list.h"

void inverse(Node *head) {
    if(head) {
        inverse(head->next);
        printf("%d ", head->data);
    }
}

int main() {
    List *list = initList();

    int n = 6;
    Node *nodes[n];
    for(int i = 0; i < n; i++) {
        nodes[i] = (Node *)malloc(sizeof(Node));
        nodes[i]->data = i;
        nodes[i]->next = NULL;
        insert(list, i, nodes[i]);
    }

    inverse(list->head->next);
    printf("\n");

    deleteList(list);
    return 0;
}